modules/apps/object/object-dynamic-data-mapping/src/main/resources/META-INF/resources/Autocomplete.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {useIsMounted} from '@liferay/frontend-js-react-web';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import {debounce} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+function getData(apiUrl, query, page, pageSize) {
+	const url = new URL(apiUrl, Liferay.ThemeDisplay.getPortalURL());
```

Comment:
There's also just `themeDisplay`. I don't know which is both, but maybe we should settle for one type of access to the object or another and enforce it as much as we can.

I took a quick look some time ago to see if we could place this elsewhere other than a big global but couldn't come up with anything of sustance.